docs/development/pigs/pig-009.rst

```diff
@@ -0,0 +1,231 @@
+proposals or study the sensitivity of future observations. For this reason, we
+propose to implement a framework for event simulation in Gammapy.
+
+The proposed framework consists of individual building blocks, represented by
+classes and methods, that can be chained together to achieve a full simulation of an event
+list corresponding to a given observation. This includes the simulation of source
+events, background events, effects of instrument response functions (IRF) and
+arrival times. As underlying method for the actual event sampling we propose to
+use inverse cumulative distribution function (CDF) sampling (inverseCDF_) with finely 
+binned discrete source and background. Temporal models will be also taken into account
+and time will be sampled separately in a 1D analysis, assuming that the temporal 
+depence of the input source models factorizes.
+
+
+Sampling methods
+====================
+Inverse CDF sampling (inverseCDF_) is an established method to sample from discrete
+probability mass functions. It is used by `ASTRIsim` (_astrisim), the event simulator 
+of the AGILE collaboration. However it is not the method of choice for other existing 
+event samplers such as the the Fermi-LAT Science Tools (gtobsim) and CTOOLS (ctobssim). 
+The latter uses a combination of analytical sampling for models, where a solution is 
+known (e.g. power-laws) and the rejection sampling method (rej_sampl_), where the sampling 
+has to be done numerically (see an example here gammalib_).
+
+As rejection sampling can directly sample from continuous probability density
+functions, it is expected to yield very precise results. However an enveloping
+distribution is needed, which should be adapted to the target distribution
+to be efficient (see also `rejection sampling in Python`_ for an example implementation),
+otherwise a lot of computation time is spend in rejecting drawn samples.
+
+For this reason we favour the inverse CDF sampling method, as a simple to implement
+and general sampling method.
+The precision of the inverse CDF sampling method can be controlled by the resolution
+of the input probability mass function (PMF) and is in practice only limited by the
+available memory. We will study the required bin-size of the PMFs to reach sufficient
+precision. If we find the inverse CDF sampling method to be not precise enough, it is 
+still possible to achieve better precision adopting the rejection sampling. This will 
+not have a strong impact on the structure of the event-sampler.
+
+
+Proposal
+========
+
+We propose to include in `gammapy.cube` an high level interface (HLI) class, labelled as
+`MapDatasetEventSampler`. This class handles the complete event sampling process, 
+including the corrections related to the IRF and source temporal variability, for a given 
+GTI / observation.
+
+As a first step, the User should create a `MapDataset` object, as described below:
+
+# 20 lines of code with the `MapDataset` object setup.
```

Comment:
@fabiopintore - What's this about 20 lines of code? Just remove this comment? Or did you want to add something?